### PR TITLE
Add multi select inspector

### PR DESCRIPTION
**Pull request by ellatrix**
This is very much a WIP, but the idea is to add an inspector for multi-select as well. This could be useful for aligning multiple blocks at once, converting multiple paragraphs to a block quote or verse form...

Still needs some thinking around how the API would look like.

### CODE REVIEW

File: editor/sidebar/multi-block-inspector/index.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from 'i18n';
```
Comment (aduth) [inline editor/sidebar/multi-block-inspector/index.js]: Note with #2172, these need to be updated to prefix the dependencies with `@wordpress/`. You will need to perform a rebase against the latest version of master and apply your changes:

```
git fetch origin
git rebase origin/master
```

**Comment by iseulde** [comment]
Rebased this one. @jasmussen So I guess instead of adding stuff to the sidebar, I'll try to keep the toolbar inline and offer the same block switcher and alignment options for multiple paragraphs.

**Comment by iseulde** [comment]
I also think we should remove the blue box on top entirely then. We can offer everything inline and more advanced things on the side. Would also be very nice on mobile with a "select" button to select multiple blocks.

**Comment by jasmussen** [comment]
Love the idea behind this 🌟 🌟 🌟 🌟 🌟 

Also agree with this:

> I also think we should remove the blue box on top entirely then. We can offer everything inline and more advanced things on the side. Would also be very nice on mobile with a "select" button to select multiple blocks.

I encountered a little flickering, though. Seemed to happen more often when the document tab was active. Tried to record a video:

flickering

**Comment by iseulde** [comment]
Strange, this should not affect the multi selection itself. Anyway, there's still some how left here. I'll move it inline. :)

**Comment by iseulde** [comment]
Redoing this whole thing so it's inline... WIP. Still need to add the formatting toolbar and allow the inspector to be the same for multi select.

**Comment by iseulde** [comment]
@mtias @aduth @youknowriad This is going to become quite messy if we keep the current API that is a single edit method returning an array and filling in slots. I would need to force the `focus` props to `true` for the first selected block and reset `setAttributes` etc. to allow updating all blocks from this one edit method.

I see two solutions:

* We have separate methods, e.g. `save()`, `edit()`, `blockControls()` and `inspectorControls()`, or something like that. 
* We have a separate method for multi-select outside the `edit()` method. This is an inferior solution though, as we're splitting it up only partially, and the block implementor needs to duplicate all inspector controls to this method if they are the same.

**Comment by iseulde** [comment]
Okay, I'm going to split this up into two or three PR so we can move forward in the meantime with other improvements.

**Comment by gziolo** [comment]
@iseulde, can you confirm that this one is still relevant?

**Comment by iseulde** [comment]
Yes! I'll redo it a bit.

**Comment by SuperGeniusZeb** [comment]
Should this be closed since #3535 exists?

**Comment by gziolo** [comment]
I'm closing it in favor of #7635 which is more up to date. Let's iterate there.

**Comment by gziolo** [comment]
Actually, it's different than  #7635. However, there is #3535 which is very similar and more up to date. In addition, we can easily reuse patterns which will come out of #7635 to apply to inspector controls, too as noted by @iseulde in this comment https://github.com/WordPress/gutenberg/pull/7635#issuecomment-402117793 :)

**Pull request closed without merging**